### ISSUE TITLE
Custom Image Template for AVD fails because of Windows Update

### ISSUE THREAD

**Issue by cjr-githu** [Azure/RDS-Templates#793]
See attached output log. Is anyone else running into issues deploying an image using the built-in custom image template in AVD?

We are seeing this same issue in mutilple tenants.

[customization_output.log](https://github.com/user-attachments/files/16331789/customization_output.log)

Is doesn't matter whether you choose the windows update script or not. Just selecting any script fails with the same windows update error.

This was using the Windows 11 Enterprise 22H2 + M365 Apps market place image. Choosing the newer 23H2 gives the same error message.  It also doesn't matter if you start the image creation from scratch or use an existing image.

I am also using the option of creating a temporary vnet for deployment instead of using an existing vnet just to avoid issues with network access blocks. Everything worked fine until this new patch tuesday cycle.

**Comment by cjr-githu** [comment]
Oh wow, thanks for the info Alex, did you follow a specific procedure to fix the issue?

> Its an error because of a windows update [rgl/packer-plugin-windows-update#144](https://github.com/rgl/packer-plugin-windows-update/issues/144)
> 
> I have the same error: https://learn.microsoft.com/en-us/windows/release-health/status-windows-11-23h2#3351msgdesc

**Comment by rcAlexterNeuzen** [comment]
> Oh wow, thanks for the info Alex, did you follow a specific procedure to fix the issue?
> 
> > Its an error because of a windows update [rgl/packer-plugin-windows-update#144](https://github.com/rgl/packer-plugin-windows-update/issues/144)
> > I have the same error: https://learn.microsoft.com/en-us/windows/release-health/status-windows-11-23h2#3351msgdesc

no i didn't solve it yet. I'm hoping there will be a recall of the windows update or something ... until then I will probably update manually a custom image..

**Comment by ewanmalone** [comment]
yes same issue here. annoyingly it happens even if I don't set windows update to run in the scripts.

**Comment by cjr-githu** [comment]
> yes same issue here. annoyingly it happens even if I don't set windows update to run in the scripts.


That is correct, I have a ticket open with MS to try and figure this out because the KIR option won't work here to roll it back.

**Comment by cjr-githu** [comment]
For anyone still running into issues with this, Microsoft has released a patch for this. It is not available yet in the marketplace image.

The patch is KB5040527 and is located at https://support.microsoft.com/en-us/topic/july-25-2024-kb5040527-os-builds-22621-3958-and-22631-3958-preview-de3e1e24-0c07-4210-9777-8e03a1446bae

For use within the custom image template, I created the script below to be used within the  own script section of the deployment.
I used a storage account with blob to upload script there.
Make sure to move the script all the way at the top of the script section and add a restart as well.

[KB5040527_Install.txt](https://github.com/user-attachments/files/16438331/KB5040527_Install.txt) (Rename the extension to .ps1)

If aynyone as questions let me know
